webots_ros2_driver/webots_ros2_driver/utils.py

```diff
@@ -135,19 +135,26 @@ def get_host_ip():
         sys.exit('Unable to get host IP address. \'ip route\' could not be executed.')
 
 
+def get_wsl_ip_address():
+    # wsl2 net mode
+    # NAT: if get_router_ip() == get_wsl_host_ip()
+    # Mirrored: if get_router_ip() != get_wsl_host_ip()
+    return get_wsl_host_ip() if get_router_ip() == get_wsl_host_ip() else '127.0.0.1'
+
+
 def controller_protocol():
     protocol = 'tcp' if (has_shared_folder() or is_wsl()) else 'ipc'
     return protocol
 
 
 def controller_ip_address():
-    ip_address = get_host_ip() if has_shared_folder() else get_wsl_ip_address()
+    ip_address = get_router_ip() if has_shared_folder() else get_wsl_ip_address()
     return ip_address
 
 
 def controller_url_prefix(port='1234'):
     if has_shared_folder() or is_wsl():
-        return 'tcp://' + (get_host_ip() if has_shared_folder() else get_wsl_ip_address()) + ':' + port + '/'
+        return 'tcp://' + (get_router_ip() if has_shared_folder() else get_wsl_ip_address()) + ':' + port + '/'
```

Comment:
```suggestion
        return 'tcp://' + (get_router_ip() if has_shared_folder() else get_wsl_host_ip()) + ':' + port + '/'
```

Reply:
but get_wsl_ip_address shoud be used here

Reply:
if the file `/etc/resolv.conf` exist, the function get_wsl_host_ip never return `127.0.0.1` whatever the mode.
But get_wsl_ip_address can return get_wsl_host_ip or localhost

Reply:
What I mean here is that in this PR, `get_wsl_ip_address` doesn't exist any more. You renamed it to `get_wsl_host_ip`. So, the above line will fail due to an attempt to call an unknown function.